core/lib/Cardano/Address/Script.hs

```diff
@@ -408,7 +546,34 @@ prettyErrKeyHashFromText = \case
     ErrKeyHashFromTextWrongDataPart ->
         "Verification key hash is Bech32-encoded but has an invalid data part."
     ErrKeyHashFromTextInvalidHex ->
-        "Invalid hex-encoded string: must be either 28, 32 or 64 bytes"
+        "Invalid hex-encoded string: must be either 28, 32 or 64 bytes."
+
+-- Possible errors when deserializing a script hash from text.
+--
+-- @since 4.0.0
+data ErrScriptHashFromText
```

Comment:
I think it would be useful to provide more details to the user when something fails, like the invalid string, or the wrong prefix, etc.